### ISSUE TITLE
Ghost notifications

### ISSUE THREAD

**Issue by jb55** [damus-io/damus#2158]
I'm noticing the notification icon going off, but when you switch to the notification tab there are no new notes.

Need a way to reproduce this...

**Comment by danieldaquino** [comment]
I am starting to see this as well, and I have a guess as to what is causing it.

I believe it is likely the Purple renewal notifications, because this issue started for me around the same time the renewal notifications started appearing. They are computed notifications, and the mechanism for registering when the notification is read/unread might not be persistent across app restarts.

What mechanism ensures that read/unread statuses on normal notes are persistent?

**Comment by jb55** [comment]
we save the last seen event's created_at in user settings. any note that has a higher created_at and matches our notification critera then the flags will be set

I believe this happens here:

```swift
/// A helper to determine if we need to notify the user of new events
func handle_last_events(new_events: NewEventsBits, ev: NostrEvent, timeline: Timeline, shouldNotify: Bool = true) -> NewEventsBits? {
    let last_ev = get_last_event(timeline)

    if last_ev == nil || last_ev!.created_at < ev.created_at {
        save_last_event(ev, timeline: timeline)
        if shouldNotify {
            return new_events.union(timeline_to_notification_bits(timeline, ev: ev))
        }
    }
    
    return nil
}
```

you can grep `shouldNotify` to find all the places that call this function

**Comment by danieldaquino** [comment]
I will work on this next!

**Comment by alltheseas** [comment]
I haven't seen this. I don't have purple, or related notifications.

**Comment by danieldaquino** [comment]
> I haven't seen this. I don't have purple, or related notifications.

Thanks for pointing this out! This further indicates that this issue is indeed specific to Purple expiry notifications

**Comment by danieldaquino** [comment]
Sent a patch for this one! https://groups.google.com/a/damus.io/g/patches/c/9cf4-qXY35M

**Comment by danieldaquino** [comment]
Talked to @jb55 and it was decided to push this fix to the upcoming v1.8 release as well.

The fix is now pushed there: 862101a
